airflow/jobs.py

```diff
@@ -435,6 +437,12 @@ def schedule_dag(self, dag, session=None):
             else:
                 next_run_date = dag.following_schedule(last_scheduled_run)
 
+            # make sure backfills are also considered
+            last_run = dag.get_last_dagrun(session=session)
+            if last_run and next_run_date:
+                while next_run_date < last_run.execution_date:
```

Comment:
nit: this looks like it should be '<=' to cover edge case (if such scenario is possible)